compiler/rustc_borrowck/src/lib.rs

```diff
@@ -110,7 +110,19 @@ fn mir_borrowck(tcx: TyCtxt<'_>, def: LocalDefId) -> &BorrowCheckResult<'_> {
     let (input_body, promoted) = tcx.mir_promoted(def);
     debug!("run query mir_borrowck: {}", tcx.def_path_str(def));
 
-    if input_body.borrow().should_skip() {
+    let input_body: &Body<'_> = &input_body.borrow();
+
+    if let Some(guar) = input_body.tainted_by_errors {
+        let result = BorrowCheckResult {
+            concrete_opaque_types: FxIndexMap::default(),
+            closure_requirements: None,
+            used_mut_upvars: SmallVec::new(),
+            tainted_by_errors: Some(guar),
+        };
+        return tcx.arena.alloc(result);
+    }
```

Comment:
The only difference between this block and the one immediately below is the `debug!`. Seems like they could be combined and the `debug!` removed or its text modified appropriately.

Reply:
the difference is the tainting

Reply:
I found a way to dedup it nicely